app/src/modules/settings/routes/extensions/extensions.vue

```diff
@@ -1,25 +1,40 @@
 <script setup lang="ts">
 import api from '@/api';
-import { ApiOutput, EXTENSION_TYPES } from '@directus/extensions';
+import { APP_OR_HYBRID_EXTENSION_TYPES, ApiOutput, ExtensionType } from '@directus/extensions';
 import { groupBy } from 'lodash';
 import { computed, ref } from 'vue';
 import { useI18n } from 'vue-i18n';
+import { useRouter } from 'vue-router';
+import { currentPageLink } from './utils/current-page-link';
+import { useReloadGuard } from './utils/use-reload-guard';
 import SettingsNavigation from '../../components/navigation.vue';
 import ExtensionGroupDivider from './components/extension-group-divider.vue';
 import ExtensionItem from './components/extension-item.vue';
 import ExtensionsInfoSidebarDetail from './components/extensions-info-sidebar-detail.vue';
+import VNotice from '@/components/v-notice.vue';
 
 const { t } = useI18n();
+const router = useRouter();
 
 const error = ref();
 const loading = ref(false);
 const extensions = ref<ApiOutput[]>([]);
+const needsReload = ref<boolean>(false);
 
 const bundled = computed(() => extensions.value.filter(({ bundle }) => !!bundle));
-const regular = computed(() => extensions.value.filter(({ bundle }) => !!bundle === false));
-
+const regular = computed(() => extensions.value.filter(({ bundle }) => !bundle));
 const extensionsByType = computed(() => groupBy(regular.value, 'schema.type'));
 
+const { confirmLeave, leaveTo, removeGuard } = useReloadGuard(needsReload);
+
+const leavePage = () => {
+	removeGuard();
+	const target = leaveTo.value ?? currentPageLink();
+	router.push(target);
+	// reload the page
```

Comment:
Could you expand this comment a tad? I'm afraid the logic here might be a bit difficult to comprehend for non-involved devs.